cranelift-codegen/src/verifier/mod.rs

```diff
@@ -1837,12 +1837,41 @@ impl<'a> Verifier<'a> {
         Ok(())
     }
 
+    fn verify_function_invalid_types(&self, errors: &mut VerifierErrors) -> VerifierStepResult<()> {
```

Comment:
Maybe call this `typecheck_function_signature` to match `typecheck_entry_block_params`?

Reply:
Done